src/Data/PostObjectType.php

```diff
@@ -0,0 +1,517 @@
+<?php
```

Comment:
I feel like this file belongs under Registry vs Data, since the goal here is to add things to the registry. 

I know there are some things in the Data directory already doing similar things (re: settings), and I think that should get moved out too. (tbh, settings need a bit of an overhaul in general, but different discussion)

Reply:
Im down for this change.

PS: I was modeling it after the `Data/{Type}Mutation classes (not settings), and breaking away from that pattern could allow us to relocate some of the extra code in `WPGraphQL.php` while still maintaining logical consistency in the class. 

Any specific [ suggestions on the naming](https://github.com/wp-graphql/wp-graphql/pull/2399#discussion_r884188865)?